test/compilable/cppmangle.d

```diff
@@ -438,3 +438,15 @@ version (linux)
 {
     static assert(test36.mangleof == "_Z6test36PPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPPiPS12_");
 }
+
+/*****************************************/
+// https://issues.dlang.org/show_bug.cgi?id=17772
+
+extern(C++, SPACE)
+int test37(T)(){ return 0;}
+
+version (Posix) // all non-Windows machines
+{
+    static assert(test37!int.mangleof == "_ZN5SPACE6test37IiEEiv");
```

Comment:
Next time we introduce an *incorrect* mangling in the test suite, please either:
- Add a comment (e.g.: "This should be `_ZN5SPACE6test37IiEET_v` but this test a partial fix for issue 17772")
- Just test the relevant part (e.g. static assert(test37!int.mangleof[0 .. 16] == "_ZN5SPACE6test37")

Introducing tests that are testing a value is wrong is just plain confusing for the people actually fixing the issue, because suddenly the test suite fails and there's no indication the test is not correct.